### DISCUSSION TITLE
Griffe 0.46.0 might be introducing odd behavior

### DISCUSSION THREAD

**Discussion by Andrew-S-Rosen** [mkdocstrings/python#169] [General]
I unfortunately don't have enough details yet to open a "proper" issue report, but in the meantime, I wanted to note that today's release of griffe 0.46.0 seems to _potentially_ introduce some unusual behavior in `mkdocstrings-python`. Namely, there are stray functions being documented in a given module. 

For instance, check out https://quantum-accelerators.github.io/quacc/reference/quacc/recipes/vasp/core.html. There are several functions here, such as `job`, `flow`, `run_and_summarize`, and `run_and_summarize_opt`, that are [not actually defined within that module](https://github.com/Quantum-Accelerators/quacc/blob/main/src/quacc/recipes/vasp/core.py) but are having their documentation shown. They do not appear with griffe<0.46.0. My next steps are to see if I can dig into what's going on...

**Comment by pawamoy** [comment]
Thanks for the speedy report, and sorry for the trouble. I'm looking into it right now.

**Reply by pawamoy** [reply]
Can you patch `griffe/mixins.py` lines 370 and 371 in your venv to see if it corrects the behavior?

```diff
-        if self.is_wildcard_exposed:
-            return _True  # type: ignore[return-value]
+        if self.parent and self.parent.is_module and bool(self.parent.exports):  # type: ignore[attr-defined]
+            return _True if self.name in self.parent.exports else _False  # type: ignore[attr-defined,return-value]
```

**Reply by pawamoy** [reply]
https://github.com/mkdocstrings/griffe/commit/ea909526f3a637849364544daff74cd49ccaf428 should fix this issue.

**Reply by Andrew-S-Rosen** [reply]
@pawamoy: Thank you so much for the quick response, especially given the fact that I have provided only bare bones information thus far. I can confirm that your patch works!

------

In case it's relevant, I am getting the following `DeprecationWarning` though:

```
INFO    -  DeprecationWarning: Passing multiple extensions as a single list or tuple is
           deprecated. Please pass them as separate arguments instead.
             File
           "/home/rosen/software/miniconda/envs/quacc/lib/python3.11/site-packages/mkdocstrings_handlers/python/handler.py",
           line 271, in collect
               extensions=load_extensions(extensions),
             File
           "/home/rosen/software/miniconda/envs/quacc/lib/python3.11/site-packages/griffe/extensions/base.py",
           line 481, in load_extensions
               warnings.warn(
INFO    -  DeprecationWarning: This method is now a property and should be accessed as
           such (without parentheses).
             File
           "/home/rosen/software/miniconda/envs/quacc/lib/python3.11/site-packages/jinja2/runtime.py",
           line 298, in call
               return __obj(*args, **kwargs)
             File
           "/home/rosen/software/miniconda/envs/quacc/lib/python3.11/site-packages/griffe/mixins.py",
           line 413, in __call__
               warnings.warn(
```

**Reply by pawamoy** [reply]
Great!

Yes, the deprecation warnings are expected and will disappear with the next version of mkdocstrings-python :)

**Reply by Andrew-S-Rosen** [reply]
Beautiful. I'll close this and want to thank you again for all you have done for this wonderful project.
